src/Models/ObjectMap/Entities/Associations/AssociationStubPolymorphic.php

```diff
@@ -41,9 +41,18 @@ public function isCompatible(AssociationStubBase $otherStub)
         $thatTarg = $otherStub->getTargType();
         $thisNull = null === $thisTarg;
         $thatNull = null === $thatTarg;
-        if ($thisNull == $thatNull) {
+        if ($thisNull == $thatNull && null === $thatNull) {
             return false;
         }
+        if((!$thisNull &&
+            !$thatNull) &&
+            ($this->getBaseType() !== $otherStub->getTargType() ||
+            $otherStub->getBaseType() !== $this->getTargType())
```

Comment:
$thisTarg = $this->getTargType();
$thatTarg = $otherStub->getTargType();

are already defned earlier in the method.  maybe define $thisBase and $thatBase off the getBaseType() results by analogy?